secrets/secrets.go

```diff
@@ -298,3 +322,34 @@ func NewSecrets(r io.Reader) (*Secrets, error) {
 	}
 	return secrets, nil
 }
+
+func csiPathParser(dir fs.FS) (Document, error) {
+	secretsDocument := Document{
+		Secrets: make(map[string]GenericSecret),
+	}
+	err := fs.WalkDir(
+		dir,
+		".",
+		func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if d.IsDir() {
+				return nil
+			}
+			// parse file
+			file, err := os.Open(path)
```

Comment:
you actually need to use `fs.Open` instead of `os.Open` (the path is relative inside the fs)

Reply:
also you did not close the file :)